packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2091,23 +2091,30 @@ class _InactiveElements {
 
   static void _deactivateRecursively(Element element) {
     assert(element._lifecycleState == _ElementLifecycle.active);
-    element.deactivate();
-    assert(element._lifecycleState == _ElementLifecycle.inactive);
-    element.visitChildren(_deactivateRecursively);
-    assert(() {
-      element.debugDeactivated();
-      return true;
-    }());
+    try {
+      element.deactivate();
+    } finally {
+      element._deactivate();
```

Comment:
Potential behavior change. Before the change you can either call `super.deactivate` wherever you want in the super implementation, but after this the framework part of the deactivation process always happen after the user callback `deactivate`.